src/common/wincmn.cpp

```diff
@@ -729,10 +715,10 @@ wxSize wxWindowBase::DoGetBestSize() const
               node = node->GetNext() )
         {
             wxWindow *win = node->GetData();
-            if ( win->IsTopLevel()
-                    || !win->IsShown()
+            if ( win->IsTopLevel() || !win->IsShown()
+                || !IsClientAreaChild(win)
 #if wxUSE_STATUSBAR
-                        || wxDynamicCast(win, wxStatusBar)
+                || wxDynamicCast(win, wxStatusBar)
```

Comment:
I think the new test makes the test for `wxStatusBar` redundant as it's only not part of the client area in `wxFrame`, which overrides `IsClientAreaChild()` to exclude it.

What bothers me is that we don't test for `wxToolBar` here, although it looks like we ought to, for the same reason. And in the new version we will exclude it, which might not be correct, if we want to use its width.